### PR TITLE
Add a Dockerfile

### PR DESCRIPTION
**Pull request by wkoszek**
I couldn't make it work on my Linux system b/c of some Python dependency problems. Made this thing run in a Docker.

### CODE REVIEW

**Comment by wkoszek** [comment]
I wonder if someone could test this ^^ for me. What I see:

```
docker build -t lgpt .
docker run -ti lgpt
```

I get to prompt and try to run the `ingest.py` and that's what I'm getting

```
# python3 ingest.py --device_type cpu
Loading documents from /root/SOURCE_DOCUMENTS
Loaded 0 documents from /root/SOURCE_DOCUMENTS
Split into 0 chunks of text
load INSTRUCTOR_Transformer
max_seq_length  512
Using embedded DuckDB with persistence: data will be stored in: /root/DB
Traceback (most recent call last):
  File "/root/ingest.py", line 78, in <module>
    main()
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/root/ingest.py", line 72, in main
    db = Chroma.from_documents(texts, embeddings, persist_directory=PERSIST_DIRECTORY, client_settings=CHROMA_SETTINGS)
  File "/usr/local/lib/python3.10/dist-packages/langchain/vectorstores/chroma.py", line 413, in from_documents
    return cls.from_texts(
  File "/usr/local/lib/python3.10/dist-packages/langchain/vectorstores/chroma.py", line 381, in from_texts
    chroma_collection.add_texts(texts=texts, metadatas=metadatas, ids=ids)
  File "/usr/local/lib/python3.10/dist-packages/langchain/vectorstores/chroma.py", line 158, in add_texts
    embeddings = self._embedding_function.embed_documents(list(texts))
  File "/usr/local/lib/python3.10/dist-packages/langchain/embeddings/huggingface.py", line 148, in embed_documents
    embeddings = self.client.encode(instruction_pairs)
  File "/usr/local/lib/python3.10/dist-packages/InstructorEmbedding/instructor.py", line 524, in encode
    if isinstance(sentences[0],list):
IndexError: list index out of range
```

The `SOURCE_DOCUMENTS/` has 1 empty file `empty.txt`.

**Review by Smmkjm — APPROVED** [review]

**Comment by oktaborg** [comment]
Is there a docker solution?

**Comment by wkoszek** [comment]
@oktaborg What do you mean?

**Comment by oktaborg** [comment]
> @oktaborg What do you mean?

Is there a docker file that I can try?

**Comment by wkoszek** [comment]
@oktaborg Yes. This branch provides this file. Just fetch my branch or go to "Files changed" tab and copy & paste the file from there.

**Comment by PromtEngineer** [comment]
> I wonder if someone could test this ^^ for me. What I see:
> 
> ```
> docker build -t lgpt .
> docker run -ti lgpt
> ```
> 
> I get to prompt and try to run the `ingest.py` and that's what I'm getting
> 
> ```
> # python3 ingest.py --device_type cpu
> Loading documents from /root/SOURCE_DOCUMENTS
> Loaded 0 documents from /root/SOURCE_DOCUMENTS
> Split into 0 chunks of text
> load INSTRUCTOR_Transformer
> max_seq_length  512
> Using embedded DuckDB with persistence: data will be stored in: /root/DB
> Traceback (most recent call last):
>   File "/root/ingest.py", line 78, in <module>
>     main()
>   File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1130, in __call__
>     return self.main(*args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1055, in main
>     rv = self.invoke(ctx)
>   File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1404, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 760, in invoke
>     return __callback(*args, **kwargs)
>   File "/root/ingest.py", line 72, in main
>     db = Chroma.from_documents(texts, embeddings, persist_directory=PERSIST_DIRECTORY, client_settings=CHROMA_SETTINGS)
>   File "/usr/local/lib/python3.10/dist-packages/langchain/vectorstores/chroma.py", line 413, in from_documents
>     return cls.from_texts(
>   File "/usr/local/lib/python3.10/dist-packages/langchain/vectorstores/chroma.py", line 381, in from_texts
>     chroma_collection.add_texts(texts=texts, metadatas=metadatas, ids=ids)
>   File "/usr/local/lib/python3.10/dist-packages/langchain/vectorstores/chroma.py", line 158, in add_texts
>     embeddings = self._embedding_function.embed_documents(list(texts))
>   File "/usr/local/lib/python3.10/dist-packages/langchain/embeddings/huggingface.py", line 148, in embed_documents
>     embeddings = self.client.encode(instruction_pairs)
>   File "/usr/local/lib/python3.10/dist-packages/InstructorEmbedding/instructor.py", line 524, in encode
>     if isinstance(sentences[0],list):
> IndexError: list index out of range
> ```
> 
> The `SOURCE_DOCUMENTS/` has 1 empty file `empty.txt`.

The error is caused because the file is empty. Probably will need to test it with some other file.

**Comment by PromtEngineer** [comment]
> I couldn't make it work on my Linux system b/c of some Python dependency problems. Made this thing run in a Docker.

Can you please update the Readme with instructions.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM ubuntu:22.04 as ubuntu
```
Comment (AhmedHathout) [inline Dockerfile:1]: I wonder if there is a reason to use Ubuntu and not just the python one like 
```suggestion
FROM python:3.11.4-slim-bullseye
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /root
+
+COPY requirements.txt ./requirements.txt
+COPY constants.py constants.py
+COPY ingest.py ingest.py
+COPY run_localGPT.py run_localGPT.py
+COPY utils.py utils.py
+COPY SOURCE_DOCUMENTS/ SOURCE_DOCUMENTS/
```
Comment (AhmedHathout) [inline Dockerfile:10]: Is it not better to have something like 
```suggestion
COPY . .
`````

And have a `.dockerignore` to ignore what is not needed? This is probably more maintainable you may be missing some important files that will be added later in the project if someone adds them but forgot to update the dockerfile.

**Pull request closed without merging**